### PR TITLE
This fixes #78170

### LINKED ISSUE #78170: Toggling a find button does not re-evaluate the search
Testing #78108

Hava a folder with two files `aa.txt`, `aaa.txt`. 

- execute dir
- search for 'aa'
-> it matches 'aaa.txt'
- now change the find toggle to `Match whole word`

I would have expected that the find is re-evaluated and `aa.txt` is matched same as in the editor.

image

### PR DESCRIPTION
**Pull request by ATheCoder**
https://github.com/microsoft/vscode/issues/78170

### CODE REVIEW

**Review by diguifi — APPROVED** [review]

**Review by Tyriar — CHANGES_REQUESTED** [review]
This doesn't quite align with the editor find as it should only trigger a new search if the current selection is no longer valid. For example, search "foo" and get a match that's wrapped in spaces, if you toggle match whole word then in the editor it won't change anything but with this PR in the terminal it will move the selection

**Comment by ATheCoder** [comment]
In the editor find, it will always move to the first match. If you select the second match and toggle any of the buttons, it will move to the first one.

**Comment by ATheCoder** [comment]
I believe right now it should work the same way as the editor find works.

File: src/vs/workbench/contrib/terminal/browser/terminalFindWidget.ts [hunk 1]
```diff
@@ -78,4 +78,14 @@ export class TerminalFindWidget extends SimpleFindWidget {
 	protected onFindInputFocusTrackerBlur() {
 		this._findInputFocused.reset();
 	}
+
+	public findFirst() {
+		const instance = this._terminalService.getActiveInstance();
+		if (instance) {
+			if (instance.hasSelection()) {
+				instance.clearSelection();
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalFindWidget.ts]: Huh, I thought the editor's find was more clever about this and wouldn't re-search from the beginning but apparently that's exactly how it works :+1:

**Review by Tyriar — APPROVED** [review]
Thanks!

**Pull request merged**